Microsoft.Research/Contracts/System.Windows.Forms/System.Windows.Forms.BindingsCollection.cs

```diff
@@ -0,0 +1,105 @@
+
+using System.Drawing;
+using System.Diagnostics.Contracts;
+using System.Windows.Forms.Layout;
+using System.ComponentModel;
+
+namespace System.Windows.Forms
+{
+  /// <summary>
+  /// Represents a collection of <see cref="T:System.Windows.Forms.Binding"/> objects for a control.
+  /// </summary>
+  public class BindingsCollection : BaseCollection
+  {
+    /// <summary>
+    /// Gets the total number of bindings in the collection.
+    /// </summary>
+    /// 
+    /// <returns>
+    /// The total number of bindings in the collection.
+    /// </returns>  
+    public override int Count
+    {
+        get
+        {
+            Contract.Ensures(Contract.Result<int>() > 0);
+            return default(int);
+        }
+    }
+
+    /// <summary>
+    /// Gets the <see cref="T:System.Windows.Forms.Binding"/> at the specified index.
+    /// </summary>
+    /// 
+    /// <returns>
+    /// The <see cref="T:System.Windows.Forms.Binding"/> at the specified index.
+    /// </returns>
+    /// <param name="index">The index of the <see cref="T:System.Windows.Forms.Binding"/> to find. </param><exception cref="T:System.IndexOutOfRangeException">The collection doesn't contain an item at the specified index. </exception><filterpriority>1</filterpriority>
+    public Binding this[int index]
+    {
+        get
+        {
+            Contract.Requires(index >= 0);
+            Contract.Requires(index < this.Count);
+            return default(Binding);
+        }
+    }  
+
+    /// <summary>
+    /// Adds the specified binding to the collection.
+    /// </summary>
+    /// <param name="binding">The <see cref="T:System.Windows.Forms.Binding"/> to add to the collection. </param>
+    protected internal void Add(Binding binding)
+    {
+        Contract.Requires(binding != null);
+    }
+
+    /// <summary>
+    /// Adds a <see cref="T:System.Windows.Forms.Binding"/> to the collection.
+    /// </summary>
+    /// <param name="dataBinding">The <see cref="T:System.Windows.Forms.Binding"/> to add to the collection.</param><exception cref="T:System.ArgumentNullException">The <paramref name="dataBinding"/> argument was null. </exception>
+    protected virtual void AddCore(Binding dataBinding)
+    {
+       Contract.Requires(dataBinding != null);
+    }
+
+    // <summary>
+    // Clears the collection of binding objects.
+    // </summary>
+    // protected internal void Clear();
+    
+    // <summary>
+    // Clears the collection of any members.
+    // </summary>
+    // protected virtual void ClearCore();
+    
+    /// <summary>
+    /// Deletes the binding from the collection at the specified index.
+    /// </summary>
+    /// <param name="index">The index of the <see cref="T:System.Windows.Forms.Binding"/> to remove. </param>
+    protected internal void RemoveAt(int index)
+    {
+        Contract.Requires(index >= 0);
+        Contract.Requires(index < this.Count);
```

Comment:
Because this method is protected internal it means that it is actually a part of the public surface. 'protected internal' means 'protected or internal' but not 'protected and internal'...

I'm ok to do this in different PR, I just pointed out that if we already adding this precondition we can add a postcondition as well...